src/app/navigation/vertical-navigation.component.ts

```diff
@@ -0,0 +1,665 @@
+  }
+
+  private addClass(className: string): void {
+    let element = this.elementRef.nativeElement;
+    this.renderer.addClass(element, className);
+  }
+
+  private removeClass(className: string): void {
+    let element = this.elementRef.nativeElement;
+    this.renderer.removeClass(element, className);
+  }
+
+  private initBodyElement(): void {
+    if (this.showBadges) {
+      this.addClass('nav-pf-vertical-with-badges');
+    }
+    if (this.persistentSecondary) {
+      this.addClass('nav-pf-persistent-secondary');
+    }
+    if (this.hiddenIcons) {
+      this.addClass('hidden-icons-pf');
+    }
+  }
+
+  private updateMobileMenu(selected?: NavigationItemConfig, secondaryItem?: NavigationItemConfig): void {
+    this.items.forEach((item) => {
+      item.isMobileItem = false;
+      if (item.children) {
+        item.children.forEach((nextSecondary) => {
+          nextSecondary.isMobileItem = false;
+        });
+      }
+    });
+
+    if (selected) {
+      selected.isMobileItem = true;
+      if (secondaryItem) {
+        secondaryItem.isMobileItem = true;
+        this.showMobileSecondary = false;
+        this.showMobileTertiary = true;
+      } else {
+        this.showMobileSecondary = true;
+        this.showMobileTertiary = false;
+      }
+    } else {
+      this.showMobileSecondary = false;
+      this.showMobileTertiary = false;
+    }
+  }
+
+   private checkNavState() {
+    window.innerWidth;
+    let width = window.innerWidth;
+
+    // Check to see if we need to enter/exit the mobile state
+    if (!this.ignoreMobile && width < this.breakpoints.tablet) {
+      if (!this.inMobileState) {
+        this.inMobileState = true;
+
+        //Set the body class to the correct state
+        this.removeClass('collapsed-nav');
+        this.addClass('hidden-nav');
+
+        // Reset the collapsed states
+        this.updateSecondaryCollapsedState(false);
+        this.updateTertiaryCollapsedState(false);
+
+        this.explicitCollapse = false;
+      }
+    } else  {
+      this.inMobileState = false;
+      this.showMobileNav = false;
+
+      // Set the body class back to the default
+      this.removeClass('hidden-nav');
+    }
+
+    if (this.explicitCollapse) {
+      this.navCollapsed = true;
+      this.addClass('collapsed-nav');
+    } else {
+      this.navCollapsed = false;
+      this.removeClass('collapsed-nav');
+    }
+  }
+
+  private collapseMenu(): void {
+    this.navCollapsed = true;
+
+    //Set the body class to the correct state
+    this.addClass('collapsed-nav');
+
+    this.explicitCollapse = true;
+  }
+
+  private expandMenu() {
+    this.navCollapsed = false;
+
+    //Set the body class to the correct state
+    this.removeClass('collapsed-nav');
+
+    this.explicitCollapse = false;
+
+    // Dispatch a resize event when showing the expanding then menu to
+    // allow content to adjust to the menu sizing
+    window.dispatchEvent(new Event('resize'));
```

Comment:
Would like to create a window service, since we're supporting mobile -- like we have for codebases in OSIO. We also reference window in the chart and action components.

"Referencing global browser objects like document or window directly from within your code is possible, but not encouraged and considered bad practice"

"Especially Angular isn’t only designed to run within your browser, but also on mobiles, the server or web workers where objects like window may not be available. Therefore the suggested approach is to wrap such objects and inject them through the dependency injection mechanism. This way it is possible to change the concrete runtime instance of a given object based on the environment the Angular application is running."